pandas/_libs/src/ujson/python/objToJSON.c

```diff
@@ -2107,6 +2114,7 @@ void Object_endTypeContext(JSOBJ Py_UNUSED(obj), JSONTypeContext *tc) {
         GET_TC(tc)->columnLabels = NULL;
 
         PyObject_Free(GET_TC(tc)->cStr);
+        free(bytes);
```

Comment:
@WillAyd Reallocating `bytes` is an outstanding issue like we discussed a while back. I know we need to free it up after we're done writing the output - but within this function bytes is undeclared, I think

Reply:
Bytes is assigned to cStr but you won’t free it in this branch. We need an extra branch that checks for the JT_BIGNUM type and frees cStr in that case